sn_networking/src/event.rs

```diff
@@ -348,18 +437,33 @@ impl SwarmDriver {
 
                     match new {
                         NatStatus::Public(_addr) => {
-                            // In theory, we could actively push our address to our peers now. But, which peers? All of them?
-                            // Or, should we just wait and let Identify do it on its own? But, what if we are not connected
-                            // to any peers anymore? (E.g., our connections timed out etc)
-                            // let all_peers: Vec<_> = self.swarm.connected_peers().cloned().collect();
-                            // self.swarm.behaviour_mut().identify.push(all_peers);
+                            info!("Relay operating in server mode");
+                            self.is_relay_client = false;                        
                         }
                         NatStatus::Private => {
-                            // We could just straight out error here. In the future we might try to activate a relay mechanism.
+                            info!("Relay operating in client mode");
+                            self.is_relay_client = true;
+
+                            for (peer_id, multiaddr) in self.relay_servers.clone() {
```

Comment:
This adds us to all relay servers we know. We might need to limit this to a certain amount.